app/models/game.rb

```diff
@@ -60,6 +61,58 @@ def initialize_board
     King.create(is_black: true, x_position: 4, y_position: 7, game_id: id, user_id: user_id_black, status: 'active')
     Queen.create(is_black: true, x_position: 3, y_position: 7, game_id: id, user_id: user_id_black, status: 'active')
   end
+
   # rubocop:enable Metrics/AbcSize
   # rubocop:enable MethodLength
+
+  # returns true if game is in check
+  def black_in_check?(x, y)
+    # identify the black kings on the board
```

Comment:
Hah!  In that case we would not want it.  I thought you could promote to anything. :P